src/brod.erl

```diff
@@ -127,12 +111,17 @@ start_link_producer(Hosts, RequiredAcks, AckTimeout) ->
 %%        time will not be included, (3) we will not terminate a
 %%        local write so if the local write time exceeds this
 %%        timeout it will not be respected.
-%%     ClientId:
-%%        Atom or binary string (preferably unique) identifier of the client.
--spec start_link_producer([host()], integer(), integer(), client_id()) ->
-        {ok, pid()} | {error, any()}.
-start_link_producer(Hosts, RequiredAcks, AckTimeout, ClientId) ->
-  brod_producer:start_link(Hosts, RequiredAcks, AckTimeout, ClientId).
+%%     max_leader_queue_len (optional, default = 32):
+%%        How many requests sent to a kafka broker can stay
+%%        unacknowledged by a broker before blocking producing thread.
+%%     max_requests_in_flight (optional, default = 5):
+%%        Worker process will be sending produce reqeusts to kafka
+%%        without waiting for acknowledgements of a previous one.
+%%        This is to control how many such requests can be sent.
+-spec start_link_producer([host()], proplists:proplist()) ->
+                             {ok, pid()} | {error, any()}.
+start_link_producer(Hosts, Options) ->
+  brod_producer:start_link(Hosts, Options).
```

Comment:
Wouldn't it be nicer to start the producers under the `brod_sup` instead of linking them to the caller process?